modules/core/test/spec/cuba/core/setget/SetGetTest.groovy

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2008-2018 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package spec.cuba.core.setget
+
+import com.haulmont.chile.core.model.MetaClass
+import com.haulmont.cuba.core.entity.StandardEntity
+import com.haulmont.cuba.core.global.Metadata
+import com.haulmont.cuba.testmodel.many2many.Many2ManyA
+import com.haulmont.cuba.testmodel.many2many.Many2ManyB
+import com.haulmont.cuba.testmodel.setget.SetGetEntity
+import com.haulmont.cuba.testsupport.TestContainer
+import org.junit.ClassRule
+import spock.lang.Shared
+import spock.lang.Specification
+
+class SetGetTest extends Specification {
+
+    @Shared
+    @ClassRule
+    public TestContainer cont = TestContainer.Common.INSTANCE
+
+    private Metadata metadata = cont.metadata()
+
+    private SetGetEntity<String> setGetEntity
+
+    void setup() {
+        setGetEntity = metadata.create(SetGetEntity.class)
+    }
+
+    def "ID and UUID"() {
+        when:
+        UUID uuid = setGetEntity.getValue("uuid")
+        setGetEntity.setValue("id", uuid)
+        UUID afterId = setGetEntity.getValue("id")
+
+        then:
+        uuid == afterId && uuid != null
+    }
+
+    def "Map"() {
+        when:
+        Map<String, Integer> map = new HashMap<>()
+        map.put("key", 12)
+        setGetEntity.setValue("map", map)
+        Map<String, Integer> afterMap = setGetEntity.getValue("map")
+
+        then:
+        map == afterMap
```

Comment:
Use reference equality for testing ?
For now you are calling equals on Map

Reply:
I intended to call equals on maps, thought it's better to use "==" instead of "equals" in groovy.

Reply:
I mean we should better check identity equality instead of == in case of direct get/set calls